### PR TITLE
fix(client): missing expiry time

### PR DESCRIPTION
**Pull request by douglaz**

### CODE REVIEW

**Review by dpc — APPROVED** [review]
Wooops.

**Review by dpc — COMMENTED** [review]
Maybe I need more coffee but since it had a bug and I still has trouble reading maybe we should either add a test, or figure out some way to make this code obviously correct.

File: modules/fedimint-ln-client/src/receive.rs [hunk 1]
```diff
@@ -205,8 +205,9 @@ impl LightningReceiveConfirmedInvoice {
                         // only when we are sure that the invoice is still pending that we can check
                         // for a timeout
                         const TOLERANCE: Duration = Duration::from_secs(60); // tolerate some clock skew
-                        let invoice_since_epoch = invoice.duration_since_epoch() + TOLERANCE;
-                        if now_since_epoch > invoice_since_epoch {
+                        let invoice_expiration_epoch =
+                            invoice.duration_since_epoch() + invoice.expiry_time() + TOLERANCE;
```
Comment (dpc) [inline modules/fedimint-ln-client/src/receive.rs:209]: **thought:** So here we add "duration" + "time" + "tolerance (duration)", so what's confusing is that (I guess, `expiry_time` is "duration", while `duration_since_epoch` is actually "(absolute) time"). :D.

File: modules/fedimint-ln-client/src/receive.rs [hunk 2]
```diff
@@ -205,8 +205,9 @@ impl LightningReceiveConfirmedInvoice {
...
+                        if now_since_epoch > invoice_expiration_epoch {
```
Comment (dpc) [inline modules/fedimint-ln-client/src/receive.rs:210]: **thought:** And now we call on3e thin `_since_epoch` and the other `_epoch` which also seems meh. :D

**Comment by dpc** [comment]
I guess we'll want a quick backport and release of this, as is makes all ln payments have 60s deadlines?

**Review by elsirion — APPROVED** [review]

File: modules/fedimint-ln-client/src/receive.rs (hunk 1 above)
Comment (dpc) [inline modules/fedimint-ln-client/src/receive.rs:209]: **thought:** BTW. If this 3 lines were in a method, than I think we could have a really tiny unit test that crafts a fake invoice and then checks against values against and after the timeout.

**Comment by fedimint-backports** [comment]
Successfully created backport PR for `releases/v0.2`:
- #3957

**Pull request merged**